src/sql/workbench/contrib/notebook/browser/selectImageDialog.ts

```diff
@@ -0,0 +1,191 @@
+import { localize } from 'vs/nls';
+import { IContextViewService } from 'vs/platform/contextview/browser/contextView';
+import * as DOM from 'vs/base/browser/dom';
+import { IClipboardService } from 'vs/platform/clipboard/common/clipboardService';
+import { IThemeService } from 'vs/platform/theme/common/themeService';
+import { ILogService } from 'vs/platform/log/common/log';
+import { ITextResourcePropertiesService } from 'vs/editor/common/services/textResourceConfigurationService';
+import { IAdsTelemetryService } from 'sql/platform/telemetry/common/telemetry';
+import { attachModalDialogStyler } from 'sql/workbench/common/styler';
+import { ILayoutService } from 'vs/platform/layout/browser/layoutService';
+import { Deferred } from 'sql/base/common/promise';
+import { InputBox } from 'sql/base/browser/ui/inputBox/inputBox';
+import * as styler from 'vs/platform/theme/common/styler';
+import { Button } from 'sql/base/browser/ui/button/button';
+import { Checkbox } from 'sql/base/browser/ui/checkbox/checkbox';
+
+export interface SelectImageOptions {
+	imageHtml: string;
+	imagePath?: string;
+	embedImage?: boolean;
+}
+
+export class SelectImageDialog extends Modal {
+	private _selectionComplete: Deferred<SelectImageOptions>;
+	private _okButton: Button;
+	private _cancelButton: Button;
+	private _imageTypeSelectBox: SelectBox;
+	private _imagePathLabel: HTMLElement;
+	private _imagePathInputBox: InputBox;
+	private _browseImagesButton: Button;
+	private _embedImageCheckbox: Checkbox;
+
+	private readonly localImageType = localize('selectImageDialog.localImageType', "Local");
+	private readonly remoteImageType = localize('selectImageDialog.remoteImageType', "Remote");
+	private readonly localImagePathLabel = localize('selectImageDialog.localPathLabel', "Image local path");
+	private readonly remoteImagePathLabel = localize('selectImageDialog.remotePathLabel', "Image remote path");
+
+	constructor(
+		@IThemeService themeService: IThemeService,
+		@ILayoutService layoutService: ILayoutService,
+		@IAdsTelemetryService telemetryService: IAdsTelemetryService,
+		@IContextKeyService contextKeyService: IContextKeyService,
+		@IContextViewService private contextViewService: IContextViewService,
+		@IClipboardService clipboardService: IClipboardService,
+		@ILogService logService: ILogService,
+		@ITextResourcePropertiesService textResourcePropertiesService: ITextResourcePropertiesService
+	) {
+		super(
+			localize('selectImageDialog.title', "Add image"),
+			TelemetryKeys.SelectImage,
+			telemetryService,
+			layoutService,
+			clipboardService,
+			themeService,
+			logService,
+			textResourcePropertiesService,
+			contextKeyService,
+			undefined);
+		this._selectionComplete = new Deferred<SelectImageOptions>();
+	}
+
+	/**
+	 * Opens the dialog and returns a promise for what options the user chooses.
+	 */
+	public open(): Promise<SelectImageOptions> {
+		this.show();
+		return this._selectionComplete.promise;
+	}
+
+	public render() {
```

Comment:
Can you attach an image of what this looks like?

Why aren't you just using the normal file picker dialog?

https://github.com/microsoft/azuredatastudio/blob/main/src/vs/platform/dialogs/common/dialogs.ts#L250

Reply:
For @corivera ...